### PR TITLE
WIP: Archive test_output folder if build 'UNSTABLE'

### PR DESCRIPTION
**Pull request by smlambert**
If tests fail, we archive the test output, which will include test results, gc logs, and other test artifacts.
[ci skip]
Signed-off-by: smlambert <slambert@gmail.com>

### CODE REVIEW

**Comment by smlambert** [comment]
@AdamBrousseau - if I wish to test a change to pipelines, do I need to setup a special build (that uses my branch)?  If so, which sandbox build is best for me to copy from when I create it?

File: buildenv/jenkins/common/test [hunk 1]
```diff
@@ -96,6 +96,12 @@ def publish() {
             junit allowEmptyResults: true, keepLongStdio: true, testResults: '**/junitreports/**/*.xml'
             step([$class: 'Publisher', reportFilenamePattern: '**/testng-results.xml'])
             step([$class: "TapPublisher", testResults: "**/*.tap"])
+            withEnv(["WORKSPACE=${WORKSPACE}", "currentBuild=${currentBuild}"]) {
```
Comment (AdamBrousseau) [inline buildenv/jenkins/common/test]: You shouldn't need this line

Comment (smlambert) [reply buildenv/jenkins/common/test]: ok, thanks, and yes, wondered as they are both global...

**Comment by smlambert** [comment]
Jenkins test sanity jdk8 plinux

**Comment by smlambert** [comment]
Jenkins test sanity plinux jdk8

**Comment by smlambert** [comment]
Jenkins test sanity plinux jdk8

**Comment by smlambert** [comment]
Jenkins test extended plinux jdk8

**Comment by smlambert** [comment]
Jenkins test extended plinux jdk8

**Comment by smlambert** [comment]
Jenkins test extended plinux jdk9

**Comment by smlambert** [comment]
@AdamBrousseau - have you seen this error before?  

groovy.lang.MissingPropertyException: No such property: test_target for class: groovy.lang.Binding
	at groovy.lang.Binding.getVariable(Binding.java:63)

I have introduced an error in extended tests to verify that the archive of test_output_xxxx folder works if UNSTABLE, but this is a script failure that is unclear to me what is causing it?  From what I can tell, test_target should be _extended.

**Comment by AdamBrousseau** [comment]
`test_target` should be `TEST_TARGET` :)

**Comment by smlambert** [comment]
Jenkins test extended plinux jdk9

**Comment by smlambert** [comment]
Jenkins test extended plinux jdk9

**Pull request closed without merging**